t/uniq.t

```diff
@@ -83,6 +83,85 @@ is_deeply( [ uniqint 6.1, 6.2, 6.3 ],
            [ 6 ],
            'uniqint compares as and returns integers' );
 
+my $ls = 31;      # maximum left shift for 32-bit unity
+
+if( $Config{ivsize} == 8 ) {
+  $ls       = 63; # maximum left shift for 64-bit unity
+}
+
+# Populate @in with UV-NV pairs of equivalent values.
+# Each of these values is exactly representable as 
+# either a UV or an NV.
+
+my @in = (1 << $ls, 2 ** $ls,
+          1 << ($ls - 3), 2 ** ($ls - 3),
+          5 << ($ls - 3), 5 * (2 ** ($ls - 3)));
+
+my @correct = (1 << $ls, 1 << ($ls - 3), 5 << ($ls -3));
+
+if( $Config{ivsize} == 8 && $Config{nvsize} == 8 ) {
+
+     # Add some more UV-NV pairs of equivalent values.
+     # Each of these values is exactly representable
+     # as either a UV or an NV.
+
+     push @in, ( 9007199254740991,     9.007199254740991e+15,
```

Comment:
This chunk has a lot of "magic numbers" in it with no suggestion of how to recreate those. It would be great if they could be generated more programmatically somehow, similar to the `@correct` case above so it's more apparent where they come from.